dot/telemetry/telemetry.go

```diff
@@ -65,9 +76,21 @@ func GetInstance() *Handler { //nolint
 				go handlerInstance.startListening()
 			})
 	}
+	if !enabled {
+		return &NoopHandler{}
+	}
+
 	return handlerInstance
 }
 
+// Initialise function to set if telemetry is enabled
+func (h *Handler) Initialise(_enabled bool) {
```

Comment:
```suggestion
func (h *Handler) Initialise(e bool) {
```
nit: we don't use leading underscore in go